apps/editing-toolkit/editing-toolkit-plugin/wpcom-global-styles/global-style-sidebar-notice.tsx

```diff
@@ -0,0 +1,55 @@
+declare const wpcomGlobalStyles: { upgradeUrl: string };
+
+import { ExternalLink, Fill, Notice } from '@wordpress/components';
+import { useSelect } from '@wordpress/data';
+import { createInterpolateElement, useEffect } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { useGlobalStylesConfig } from './use-global-styles-config';
+import { recordUpgradeNoticeClick, recordUpgradeNoticeShow } from './utils';
+
+const GLOBAL_STYLES_SIDEBAR = 'edit-site/global-styles';
+
+export function GlobalStylesSidebarNotice() {
+	const area = useSelect(
+		// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+		// @ts-ignore
+		( select ) => select( 'core/interface' ).getActiveComplementaryArea( 'core/edit-site' ),
+		[]
+	);
+
+	const isGlobalStylesSidebar = GLOBAL_STYLES_SIDEBAR === area;
+
+	const isVisible = useGlobalStylesConfig().isVisible;
+
+	useEffect( () => {
+		if ( isVisible && isGlobalStylesSidebar ) {
+			recordUpgradeNoticeShow( GLOBAL_STYLES_SIDEBAR );
+		}
+	}, [ isVisible, isGlobalStylesSidebar ] );
+
+	return (
+		<Fill name="ComplementaryArea/core/edit-site">
+			{ /*
+			We'll need to do the condition here because if we are doing an early return, the fill will be introduced at the bottom of the page, which means some additional CSS magic needs to be done.
+			*/ }
+			{ !! isVisible && isGlobalStylesSidebar && (
```

Comment:
Isn't `isVisible` already a boolean? If so, you shouldn't need to cast it with `!!`.

Reply:
It needed to be typecasted in the use function. This should simplify a bit the condition.